src/logger.rs

```diff
@@ -0,0 +1,100 @@
+use std::fs::File;
+use std::io::Write;
+use std::time::SystemTime;
+
+use anyhow::{Error, Result};
+use chrono::prelude::*;
+
+use crate::paths;
+
+pub struct Logger {
+    file: Option<File>,
+    error: Option<Error>,
+    level: i8,
+}
```

Comment:
I don't like the fact that this permits 4 states:
no logger and no error, both logger and error, only logger and no error, only error and no logger
more natural way to model it would be
```rs
pub struct Logger(InnerLogger);

enum InnerLogger {
  LogToFile(File),
  CannotInitialize(Error)
}
```
(`Logger` only being separated from `InnerLogger` to prohibit pattern matching)